src/server/engine_factory.rs

```diff
@@ -171,8 +173,35 @@ impl<ER: RaftEngine> KvEngineFactory<ER> {
 
 impl<ER: RaftEngine> TabletFactory<RocksEngine> for KvEngineFactory<ER> {
     #[inline]
-    fn create_tablet(&self) -> Result<RocksEngine> {
+    fn create_root_db(&self) -> Result<RocksEngine> {
         let root_path = self.kv_engine_path();
-        self.create_tablet(&root_path)
+        let tablet = self.create_tablet(&root_path)?;
+        let mut root_db = self.inner.root_db.lock().unwrap();
```

Comment:
I don't get it. For v1, it's pointless or even wrong to call create_root_db multiple times.

Reply:
The whole point of abstraction is to unify the db creation API so that the caller don't need to care if it's a single db or tablet. Under the hood, the root_db (shared_db) is only created once in v1. 

Otherwise what's the point of Trait TabletFactory.

Reply:
> I don't get it. For v1, it's pointless or even wrong to call create_root_db multiple times.

For create_shared_db, the behavior is same. It will fail if created by multiple times. 
But here the change is just to cache the clone, so that next time for create_tablet, it can return it directly---essentially it has the same API with v2, but with v1's implementation. The purpose is to potentially share the code. If in the end it's not used, I will remove it. But I believe it's helpful, at least in some test cases.